### PR TITLE
add pre-commit hook to test the coverage

### PR DESCRIPTION
**Pull request by jyenes**
adds a pre-commit hook, that avoids performing any commit if the coverage gets down or the tests are broken

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -171,5 +180,8 @@
     "webpack": "^3.8.1",
     "webpack-dev-middleware": "^3.1.3",
     "webpack-hot-middleware": "^2.22.1"
-  }
+  },
+  "pre-commit": [
+    "test"
+  ]
```
Comment (hem-brahmbhatt) [inline package.json]: Not sure about running tests before every commit... I can see that getting in the way while developing. It should be enough if it fails the build on CI.

Comment (diversemix) [reply package.json]: I agree... I've commented here https://github.com/elifesciences/elife-xpub/issues/1462#issuecomment-463545996

Comment (jyenes) [reply package.json]: we can do it in pre-push, but I think is necessary before to upload new code, to run the tests and be sure there are no tests broken

File: package.json [hunk 2]
```diff
@@ -64,6 +64,14 @@
       "<rootDir>/{client,app,server}/**/*.js",
       "!**/*.{config,test}.js"
     ],
+    "coverageThreshold": {
```
Comment (diversemix) [inline package.json:67]: What does this achieve?

Comment (jyenes) [reply package.json:67]: this is the way to configure jest, with a threshold, so if the coverage goes down to this threshold the test will fail

File: package.json (hunk 1 above)
Comment (jyenes) [reply package.json]: removed pre-commit, with this config, now the test will be executed on the CI, and will fail if the coverage decrease, from the one defined in the threshold, which matches with our current coverage.
the idea behind this is to update the threshold config periodically, so will ensure the coverage increases

**Comment by jyenes** [comment]
removed pre-commit, with this config, now the test will be executed on the CI, and will fail if the coverage decrease, from the one defined in the threshold, which matches with our current coverage.
the idea behind this is to update the threshold config periodically, so will ensure the coverage increases

File: package.json (hunk 2 above)
Comment (diversemix) [reply package.json:67]: ok - so how does this achieve not being able to decrease coverage? Do we have to continually update this ?

Comment (hem-brahmbhatt) [reply package.json:67]: Yes

Comment (hem-brahmbhatt) [reply package.json:67]: We can automate the coverage threshold in a separate issue.

**Review by hem-brahmbhatt — APPROVED** [review]

**Review by diversemix — APPROVED** [review]

**Pull request merged**